masonry/src/widget/widget_ref.rs

```diff
@@ -168,38 +161,39 @@ impl<'w> WidgetRef<'w, dyn Widget> {
         }
     }
 
-    /// Recursively find innermost widget at given position.
-    ///
-    /// If multiple overlapping children of a widget contain the given position in their layout
-    /// boxes, the last child as determined by [`Widget::children_ids`] is chosen.
+    /// Recursively find the innermost widget at the given position, using
+    /// [`Widget::get_child_at_pos`] to descend the widget tree. If `self` does not contain the
+    /// given position in its layout rect or clip path, this returns `None`.
     ///
-    /// **pos** - the position in local coordinates (zero being the top-left of the
-    /// inner widget).
-    pub fn find_widget_at_pos(&self, pos: Point) -> Option<WidgetRef<'w, dyn Widget>> {
-        let mut innermost_widget: WidgetRef<'w, dyn Widget> = *self;
-
-        if !self.state().layout_rect().contains(pos) {
+    /// **pos** - the position in global coordinates (e.g. `(0,0)` is the top-left corner of the
+    /// window).
+    pub fn find_widget_at_pos<'a>(
+        &self,
+        root: &'a RenderRoot,
+        pos: Point,
+    ) -> Option<WidgetRef<'a, dyn Widget>> {
+        // Get self from the widget arena to bind it to the arena's lifetime. Is there a way around
+        // this? Also see the comment inside the loop rebinding child to the arena's lifetime.
+        let mut innermost_widget = root.get_widget(self.id()).unwrap();
+
+        let relative_pos = pos - self.state().window_origin().to_vec2();
+        if !self.state().window_layout_rect().contains(pos)
+            || !self
+                .state()
+                .clip
+                .map_or(true, |clip| clip.contains(relative_pos))
+        {
             return None;
         }
```

Comment:
```suggestion
        if !self.state().layout_rect().contains(relative_pos) {
            return None;
        }
        if !self.state().clip.unwrap_or_default().contains(relative_pos) {
            return None;
        }
```

Reply:
Clip path as `Rect` defaults to (0,0,0,0), so unwrap_or_default doesn't work as expected. We'd need something like `Rect::INFINITY`.

Reply:
Did split into two statements